accname/name/comp_labelledby_hidden_nodes.html

```diff
@@ -0,0 +1,266 @@
+    <span id="b11" style="visibility: hidden;">
+        foo
+        <span id="b12" style="visibility: hidden;">bar</span>
+    </span>
+
+    <button aria-labelledby="b21" data-expectedlabel="foo bar baz" data-testname="button with aria-labelledby using visibility:hidden hidden span (with nested visibility:hidden spans, depth 2)" class="ex"></button>>x</button>
+    <span id="b21" style="visibility: hidden;">
+        foo
+        <span id="b22" style="visibility: hidden;">
+            bar
+        <span id="b23" style="visibility: hidden;">baz</span>
+        </span>
+    </span>
+
+    <button aria-labelledby="b31" data-expectedlabel="foo" data-testname="button with aria-labelledby using span without visibility:hidden (with nested visibility:hidden spans, depth 2)" class="ex"></button>>x</button>
+    <span id="b31">
+        foo
+        <span id="b32" style="visibility: hidden;">
+            bar
+        <span id="b33" style="visibility: hidden;">baz</span>
+        </span>
+    </span>
+
+    <button aria-labelledby="b41" data-expectedlabel="foo bar baz" data-testname="button with aria-labelledby using visibility:hidden hidden span (with nested visibility:hidden sibling spans)" class="ex"></button>>x</button>
+    <span id="b41" style="visibility: hidden;">
+        foo
+        <span id="b42" style="visibility: hidden;">bar</span>
+        <span id="b43" style="visibility: hidden;">baz</span>
+    </span>
+
+    <button aria-labelledby="b51" data-expectedlabel="foo" data-testname="button with aria-labelledby using span without visibility:hidden (with nested visibility:hidden sibling spans)" class="ex"></button>>x</button>
+    <span id="b51">
+        foo
+        <span id="b52" style="visibility: hidden;">bar</span>
+        <span id="b53" style="visibility: hidden;">baz</span>
+    </span>
+
+    <button aria-labelledby="b61" data-expectedlabel="foo bar baz" data-testname="button with aria-labelledby using span with visibility:hidden (with nested visible sibling spans)" class="ex"></button>>x</button>
+    <span id="b61" style="visibility: hidden;">
+        foo
+        <span id="b62">bar</span>
+        <span id="b63">baz</span>
+    </span>
+
+<h2>Testing with <code>visibility:collapse</code></h2>
+
+    <button aria-labelledby="c11" data-expectedlabel="foo bar" data-testname="button with aria-labelledby using visibility:collapse span (with nested span)" class="ex"></button>>x</button>
+    <span id="c11" style="visibility: collapse;">
+        foo
+        <span id="c12" style="visibility: collapse;">bar</span>
+    </span>
+
+    <button aria-labelledby="c21" data-expectedlabel="foo bar baz" data-testname="button with aria-labelledby using visibility:collapse hidden span (with nested visibility:collapse spans, depth 2)" class="ex"></button>>x</button>
+    <span id="c21" style="visibility: collapse;">
+        foo
+        <span id="c22" style="visibility: collapse;">
```

Comment:
I think this collapse in collapse is also redundant because in either case, the interior element is not rendered, so hidden. Ditto for several more below.